### ISSUE TITLE
Add `Proxy` configuration group to `OidcTenantConfig`

### ISSUE THREAD

**Issue by sberyozkin** [quarkusio/quarkus#8088]
**Description**
Proxy options may need to be set for the OIDC connection to be created.

**Implementation ideas**
Paulo explained that Vertx client will use [ProxyOptions](https://vertx.io/docs/apidocs/io/vertx/core/net/ProxyOptions.html) if they are set.

Proposal: add `OidcTenantConfig` `Proxy` configuration group and have 4 optional properties there for a start: `host`, `port`, `username`, `password`:
```
quarkus.oidc.proxy.host=123.345.456.789
quarkus.oidc.proxy.port=5678
```

**Comment by devisions** [comment]
@sberyozkin As discussed on the chat, as we have this need on our case, I'll try to come back with a PR. Thanks again for the feedback.

**Comment by sberyozkin** [comment]
@devisions Thanks, perhaps it can make it to 1.3.1.Final then

**Comment by devisions** [comment]
@sberyozkin [PR 8120](https://github.com/quarkusio/quarkus/pull/8120) contains the additions.
I am waiting for any review and feedback. As being my first contribution to this project, maybe I am missing something.

After this, I would like to extend the existing blog entry to showcase this addition also. But that's another story that we can discuss right after we complete this topic. 😊

**Comment by devisions** [comment]
It would be great to include it in the upcoming 1.3.1 whose milestone has due date this Friday.

**Comment by devisions** [comment]
@sberyozkin Sergey, just a small update that reflects our final decision is:
- to just provide the "host" (as `quarkus.oidc.proxy.host` config item)
- "port" has the default value of 80 (if not explicitly configured)
- of course, "username" and "password" are optional, since not all proxies require authn.

**Comment by sberyozkin** [comment]
@devisions Sorry for a delay, I actually missed this comment. I commented earlier at the PR that it was considered to be a new feature and we could only apply the bug fixes to 1.3.1.Final. 
Sorry, `1.4.0` won't be in the works for long :-)
